### ISSUE TITLE
vmalert sets incorrect time for alert endsAt for alertmanager notifications.

### ISSUE THREAD

**Issue by f41gh7** [VictoriaMetrics/VictoriaMetrics#5995]
### Describe the bug

 currently, vmalert truncates evalDelay from evalution timestamp for alerts. 

Eval delay has 30s as default value.

Alert resolve timestamp has the following formula for calculation = evalTimestamp + 4 * groupInterval.

For groupInterval=5s, resolve time is evalTimestamp*20s. 

But if 30s subtracted  from it, result resolveTimestamp ll be evalTimestamp*-10s. Which immediately resolves alert and it never ll result in notification at alertmanager.

 Current mitigation, set rule.evalDelay=0

### To Reproduce

 Create simple alerting rule:
```
groups:
  - name: examples
    concurrency: 1
    interval: 5s
    rules:
      - alert: always-firing
        expr: vector(1) > 0 
        for: 20s
        labels:
          severity: critical
        annotations:
          summary: "rule must be always at firing state"
```
 And check alerts at alertmanager.

### Version

Introduced at https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5155

**Comment by hagen1778** [comment]
@f41gh7 https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6028 has been merged and will be available in the next release.

**Comment by valyala** [comment]
The bugfix for this issue has been included in [vmalert v1.100.0](https://docs.victoriametrics.com/changelog/). Closing the issue as fixed.

**Comment by valyala** [comment]
The bugfix has been included in [vmalert v1.93.14 LTS](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.93.14).

**Comment by valyala** [comment]
The bugfix has been also included in [v1.97.4 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.97.4).
